packages/trader/src/AppV2/Components/TradeParameters/Strike/__tests__/strike.spec.tsx

```diff
@@ -0,0 +1,139 @@
+            <ul>
+                {data.map(({ value }: { value: string }) => (
+                    <li key={value}>
+                        <button onClick={() => setSelectedValue(value)}>{value}</button>
+                    </li>
+                ))}
+            </ul>
+        </div>
+    )),
+}));
+
+jest.mock('lodash/debounce', () => (fn: { cancel: () => void }) => {
+    fn.cancel = jest.fn();
+    return fn;
+});
+
+describe('Strike', () => {
+    let default_mock_store: ReturnType<typeof mockStore>;
+
+    beforeEach(
+        () =>
+            (default_mock_store = mockStore({
+                modules: {
+                    trade: {
+                        ...mockStore({}),
+                        barrier_1: '+1.80',
+                        barrier_choices: ['+1.80', '+1.00', '+0.00', '-1.00', '-1.80'],
+                        contract_type: TRADE_TYPES.VANILLA.CALL,
+                        currency: 'USD',
+                        proposal_info: {
+                            [CONTRACT_TYPES.VANILLA.CALL]: { obj_contract_basis: { value: '14.245555' } },
+                        },
+                    },
+                },
+            }))
+    );
+
+    afterEach(() => jest.clearAllMocks());
+
+    const mockStrike = () =>
+        render(
+            <TraderProviders store={default_mock_store}>
+                <ModulesProvider store={default_mock_store}>
+                    <Strike is_minimized />
+                </ModulesProvider>
+            </TraderProviders>
+        );
+    it('should render Skeleton loader if strike (barrier_1) is falsy', () => {
+        default_mock_store.modules.trade.barrier_1 = '';
+        mockStrike();
+
+        expect(screen.getByTestId('dt_skeleton')).toBeInTheDocument();
+        expect(screen.queryByText(strike_trade_param_label)).not.toBeInTheDocument();
+    });
+
+    it('should render trade param with "Strike price" label and input with value equal to current strike value (barrier_1)', () => {
+        mockStrike();
+
+        expect(screen.getByText(strike_trade_param_label)).toBeInTheDocument();
+        expect(screen.getByRole('textbox')).toHaveValue('+1.80');
+    });
+
+    it('should open ActionSheet with WheelPicker component, Payout per point information, "Save" button and text content with definition if user clicks on trade param', () => {
+        mockStrike();
+
+        expect(screen.queryByTestId('dt-actionsheet-overlay')).not.toBeInTheDocument();
+
+        userEvent.click(screen.getByText(strike_trade_param_label));
+
+        expect(screen.getByTestId('dt-actionsheet-overlay')).toBeInTheDocument();
+        expect(screen.getByText('WheelPicker')).toBeInTheDocument();
+        expect(screen.getByText('Payout per point:')).toBeInTheDocument();
+        expect(screen.getByText(/14.245555/)).toBeInTheDocument();
+        expect(screen.getByText('Save')).toBeInTheDocument();
+        expect(screen.getByText('Content goes here.')).toBeInTheDocument();
```

Comment:
maybe we could check that it's invisible if possible?

Reply:
I will reuse your TradeParamDefinition component after your PR will be merged, thank you👍